tracking/tracking_tests/reporter_tests.cpp

```diff
@@ -0,0 +1,62 @@
+#include "base/thread.hpp"
+#include "coding/traffic.hpp"
+
+#include "platform/location.hpp"
+#include "platform/socket.hpp"
+
+#include "std/cmath.hpp"
+
+#include "testing/testing.hpp"
+
+#include "tracking/reporter.hpp"
+
+namespace
+{
+template <class Condition>
+bool WaitCondition(Condition condition, size_t durationMs = 1000)
```

Comment:
All what you need is just implement a blocking ReadServer() method. All what you need for that is a single conditional variable.
Also, I'm not sure that this test is correct - what if there will be two calls of TestSocket::Write()? I suggest you to destroy the reporter and check that socked is closed before call of ReadServer().